js_compiler/compile.js

```diff
@@ -307,16 +406,13 @@ function Module() {
 	this.outputFunctionSection = function (out) {
 		var me = this;
 		outputUInt32(out, 1); // the section code.
-		outputUInt32(out, this.functions.bodies.length);
-		this.functions.forEach(function (name, index, body) {
-			me.outputFunction(out, name, index, body);
-		});
-	}
-
-	this.outputFunction = function (out, name, index, body) {
-		outputString(out, name);
-		outputUInt32(out, index);
-		body.output(out);
+		outputUInt32(out, this.functions.length);
+		for (var i = 0; i < this.functions.length; ++i) {
+			var func = this.functions[i];
+			outputString(out, func.name);
+			outputUInt32(out, func.index);
+			func.output(out);
+		}
```

Comment:
Was this just a simplification of the old code? Is the behaviour here the same? (It looks like it is)

Reply:
Yup. Basically, manual inlining.